core/src/test/scala/mill/main/ParseArgsTest.scala

```diff
@@ -0,0 +1,290 @@
+package mill.main
+
+import mill.discover.Mirror
+import utest._
+
+object ParseArgsTest extends TestSuite {
```

Comment:
Could we add a test to make sure nested `{}`s work:

```bash
$ echo {hello,world.{cow,moo}}
hello world.cow world.moo
```

And `{}`s within the cross-version brackets:

```bash
$ echo bridges[{2.11.11,2.11.8}].scalaVersion
bridges[2.11.11].scalaVersion bridges[2.11.8].scalaVersion
```

They currently work in Bash, not sure if they work in the parser you implemented, but even if they work we should test and verify.

Reply:
The latter works fine, nested `{}` doesn't work as expected yet